src/modules/site-deal/site-deal.service.ts

```diff
@@ -0,0 +1,118 @@
+import { CreateFacilityFolderResponseDto } from './dto/create-facility-folder-response.dto';
+import { SiteDealFolderNotFoundException } from './exception/site-deal-folder-not-found.exception';
+
+type RequiredSharepointConfigKeys = 'tfisFacilityListId' | 'tfisSharepointUrl' | 'scSharepointUrl' | 'scSiteFullUrl' | 'tfisFacilityHiddenListTermStoreId';
+type RequiredCustodianConfigKeys = 'facilityTemplateId' | 'facilityTypeGuid';
+@Injectable()
+export class SiteDealService {
+  constructor(
+    @Inject(SharepointConfig.KEY)
+    private readonly sharepointConfig: Pick<ConfigType<typeof SharepointConfig>, RequiredSharepointConfigKeys>,
+    @Inject(CustodianConfig.KEY)
+    private readonly custodianConfig: Pick<ConfigType<typeof CustodianConfig>, RequiredCustodianConfigKeys>,
+    private readonly graphService: GraphService,
+    private readonly custodianService: CustodianService,
+  ) {}
+
+  async createFacilityFolder(
+    siteId: UkefSiteId,
+    dealId: UkefId,
+    createFacilityFolderRequestItem: CreateFacilityFolderRequestItem,
+  ): Promise<CreateFacilityFolderResponseDto> {
+    const { facilityIdentifier, buyerName } = createFacilityFolderRequestItem;
+
+    const parentFolderName = this.getParentFolderName(buyerName, dealId);
+
+    const facilityFolderName = this.getFacilityFolderName(facilityIdentifier);
+    const parentFolderId = await this.getParentFolderId(siteId, parentFolderName);
+    const termGuid = await this.getTermGuid(facilityIdentifier);
+    const termTitle = facilityIdentifier;
+
+    // TODO apim-139: It's worth noting here that we don't use destinationMarket or riskMarket at all
+    const custodianCreateAndProvisionRequest = this.createCustodianCreateAndProvisionRequest(facilityFolderName, parentFolderId, termGuid, termTitle);
+
+    await this.custodianService.createAndProvision(custodianCreateAndProvisionRequest);
+
+    return {
+      folderName: facilityFolderName,
+    };
+  }
+
+  private getFacilityFolderName(facilityIdentifier: string): string {
+    return `F ${facilityIdentifier}`;
+  }
+
+  private getParentFolderName(buyerName: string, dealId: string): string {
+    return `${buyerName}/D ${dealId}`;
+  }
+
+  private async getParentFolderId(siteId: UkefSiteId, parentFolderName: string): Promise<number> {
+    const parentFolderData: GraphGetListItemsResponseDto = await this.graphService.get({
+      path: `${this.sharepointConfig.scSharepointUrl}:/lists/${this.sharepointConfig.tfisFacilityListId}/items`,
+      filter: `fields/ServerRelativeUrl eq '/sites/${siteId}/CaseLibrary/${parentFolderName}'`,
+      expand: 'fields($select=Title,ServerRelativeUrl,Code,ID,ParentCode)',
+    });
+
+    if (!parentFolderData.value.length) {
+      throw new SiteDealFolderNotFoundException(
+        `Site deal folder not found: ${parentFolderName}. Once requested, in normal operation, it will take 5 seconds to create the deal folder`,
+      );
+    }
+
+    return parseInt(parentFolderData.value[0].fields.id);
```

Comment:
Do we need error handling for if the value is not a number, or if there is no `id` response field? (and similar questions for the other parts of this file that read data from list items)